### ISSUE TITLE
false positives on svelte snippets

### ISSUE THREAD

**Issue by tborychowski** [harshmandan/svelte-check-native#10]
`svelte-check-native` shows errors on the following code:

```svelte
{#snippet mySnippet(name)}
	hello {name}
{/snippet}

{@render snip(name)}

<script>
	const name = 'john';
	let snip = mySnippet;
</script>
```

the error is:
```
Error: Cannot find name 'mySnippet'. (TS2304)
 8 | 	const name = 'john';
 9 | 	let snip = mySnippet;
     	           ^^^^^^^^^
10 | </script>
```

This is a [valid svelte code](https://svelte.dev/playground/e68af30e90fb46389bbdb8fc6e588b77?version=latest), and `svelte-check` also passes with no errors.
